### PR TITLE
Increased precision of decimal formatter

### PR DESCRIPTION
**Pull request by minecraft7771**
Fixes that the paramter popup has too low precision for some applications.

### CODE REVIEW

File: src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public class BaseTextFieldWidget extends Widget implements IWidgetParent, Intera
 
     static {
         format.setGroupingUsed(false);
+        format.setMaximumFractionDigits(8);
```
Comment (miozune) [inline src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java:44]: Wouldn't it make sense to allow configuring digits per widget? I mean adding setter method

**Review by miozune — COMMENTED** [review]
Thanks, I didn't notice this exists

File: src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java (hunk 1 above)
Comment (minecraft7771) [reply src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java:44]: Is it possible to rework the formatter to be non static for this?

Comment (miozune) [reply src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java:44]: Oh I forgot about that. Deprecate static one and add non-static variable `decimalFormat`

Comment (minecraft7771) [reply src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java:49]: Some methods reference `format` from a static context so I left them untouched. However I tested in TT to see if I can set the maximum fraction digits and that worked.

File: src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java [hunk 2]
```diff
@@ -359,6 +367,22 @@ public BaseTextFieldWidget setFocusOnGuiOpen(boolean focused) {
...
     }
 
+    public BaseTextFieldWidget setMaximumFractionDigits(int digits) {
+        if (decimalFormat != null) {
+            decimalFormat.setMaximumFractionDigits(digits);
+        }
+        return this;
+    }
+
+    public BaseTextFieldWidget setDecimalFormatter(DecimalFormat decimalFormat) {
```
Comment (miozune) [inline src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java:377]: Make this arg `NotNull`

File: src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java [hunk 3]
```diff
@@ -359,6 +367,22 @@ public BaseTextFieldWidget setFocusOnGuiOpen(boolean focused) {
         return this;
     }
 
+    public BaseTextFieldWidget setMaximumFractionDigits(int digits) {
+        if (decimalFormat != null) {
```
Comment (miozune) [inline src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java:371]: then this null check is effectively unneeded

Comment (miozune) [reply src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java:371]: unresolved

Comment (minecraft7771) [reply src/main/java/com/gtnewhorizons/modularui/common/widget/textfield/BaseTextFieldWidget.java]: My bad

**Review by miozune — APPROVED** [review]
Thanks!

**Pull request merged**